### DISCUSSION TITLE
RBAC: ACLs and user groups

### DISCUSSION THREAD

**Discussion by tomchop** [yeti-platform/yeti#927] [General]
Yeti currently supports having multiple users, as well users with super prvileges (can reset other user's passwords, etc.). This is not currently used anywhere, except to have people authenticate to the service.

https://github.com/yeti-platform/yeti/issues/916 surfaces the need for auditability, which would also include tracking user's actions on the platform.

We should also consider whether we want to add ACLs, and if so, how we want this implemented. Individual-level ACLs, group-level ACLs?

**Comment by udgover** [comment]
Group-level ACLs would be really useful! I've in mind at least the following ones:
* Admin role, self explanatory :)
* Standard role, give permissions to add data
* Read only role, can browse / query all the data but can't modify / add

**Comment by udgover** [comment]
After talking about the implementation details, we could probably pick some inspiration from [OpenRelik](https://github.com/openrelik/openrelik-server/pull/31)

**Comment by tomchop** [comment]
Beyond these roles

* admin
* read/write
* read-only

Do we also want to implement visibility on objects? (e.g. an object is only visible to certain groups). How we approach this requires more discussion I think, so maybe as a second step?

**Comment by udgover** [comment]
Implemented in https://github.com/yeti-platform/yeti/pull/1214
